pkg/api/controller.go

```diff
@@ -378,6 +384,14 @@ func (c *Controller) LinkPhysicalAddress(w http.ResponseWriter, r *http.Request,
 	writeTime := time.Now()
 	physicalAddress, addressType := normalizePhysicalAddress(repo.StorageNamespace, StringValue(body.Staging.PhysicalAddress))
 
+	// validate token
+	err = c.Catalog.GetAddressToken(ctx, repository, physicalAddress)
+	if err != nil {
+		writeError(w, r, http.StatusInternalServerError, err)
+		return
+	}
```

Comment:
We should delete the token we are currently using, before we try create an entry

Reply:
This will ensure we use the token only once

Reply:
GetAddressToken also deletes the token

Reply:
I don't think that this is a good behavior. We should call deleteToken explicitly wherever required